EngineLayer/FdrAnalysis/FdrAnalysisEngine.cs

```diff
@@ -207,55 +156,32 @@ private void DoFalseDiscoveryRateAnalysis(FdrAnalysisResults myAnalysisResults)
             if (AnalysisType == "PSM")
             {
                 CountPsm();
+                if (AllPsms.Count > 0)
+                {
+                    myAnalysisResults.BinarySearchTreeMetrics = PEP_Analysis.ComputePEPValuesForAllPSMsGeneric(AllPsms);
+                    Compute_PEPValue_Based_QValue(AllPsms);
+                }
             }
-        }
-
-        private static double GetEValue(PeptideSpectralMatch psm, int globalMeanCount, double globalMeanScore, out double maximumLikelihood)
-        {
-            // get all of the PSM's scores for all hits, sort them, then remove the last value (the best score)
-            List<double> scoresWithoutBestHit = new List<double>();
-            scoresWithoutBestHit.AddRange(psm.AllScores);
-            scoresWithoutBestHit.Sort();
-
-            if (scoresWithoutBestHit.Any())
-            {
-                scoresWithoutBestHit.RemoveAt(scoresWithoutBestHit.Count - 1);
-            }
-
-            // this is the "default" case for when there are no scores except the best hit
-            // it uses a global mean score (all scores across all PSMs) to generate the null Poisson distribution
-            // this will be overriden by the next few lines if there are enough scores in this PSM to estimate a null distribution
-            double preValue = SpecialFunctions.GammaLowerRegularized(globalMeanScore, psm.Score);
-            maximumLikelihood = globalMeanScore;
 
-            // calculate single-spectrum evalue if there are enough hits besides the best scoring peptide
-            if (psm.Score == 0)
+            if (AnalysisType == "Peptide")
```

Comment:
This is not asking for changes. I just feel like the ML input of 'Peptide' should be different from 'PSM'. Maybe adding more feature such as retention time evaluation of psms with the same peptide.